howdoi/howdoi.py

```diff
@@ -383,17 +412,21 @@ def howdoi(raw_query):
 
     res = cache.get(cache_key)
```

Comment:
After the updates I suggested above, we will be storing `answers` in the cache. We'll simply pull them out of the cache and then call `_format_answers`.

I wonder if we need to version the cache key as we make these larger changes to it. If someone has an existing cache key (which used to store strings) and now we're storing objects, we'll have issues. 

A function like `_get_cache_key` which takes a string and returns a cache key string would be useful here, as this function can deal with adding the version number.